packages/client/src/devices/MicrophoneManager.ts

```diff
@@ -16,37 +17,48 @@ import { createSubscription } from '../store/rxUtils';
 import { RNSpeechDetector } from '../helpers/RNSpeechDetector';
 
 export class MicrophoneManager extends InputMediaDeviceManager<MicrophoneManagerState> {
+  private speakingWhileMutedNotificationEnabled = true;
   private soundDetectorCleanup?: Function;
   private rnSpeechDetector: RNSpeechDetector | undefined;
   private noiseCancellation: INoiseCancellation | undefined;
   private noiseCancellationChangeUnsubscribe: (() => void) | undefined;
   private noiseCancellationRegistration?: Promise<() => Promise<void>>;
 
-  constructor(call: Call) {
-    super(call, new MicrophoneManagerState(), TrackType.AUDIO);
-
-    combineLatest([
-      this.call.state.callingState$,
-      this.call.state.ownCapabilities$,
-      this.state.selectedDevice$,
-      this.state.status$,
-    ]).subscribe(async ([callingState, ownCapabilities, deviceId, status]) => {
-      if (callingState !== CallingState.JOINED) {
-        if (callingState === CallingState.LEFT) {
-          await this.stopSpeakingWhileMutedDetection();
-        }
-        return;
-      }
-      if (ownCapabilities.includes(OwnCapability.SEND_AUDIO)) {
-        if (status === 'disabled') {
-          await this.startSpeakingWhileMutedDetection(deviceId);
-        } else {
-          await this.stopSpeakingWhileMutedDetection();
-        }
-      } else {
-        await this.stopSpeakingWhileMutedDetection();
-      }
-    });
+  constructor(
+    call: Call,
+    disableMode: TrackDisableMode = isReactNative()
+      ? 'disable-tracks'
+      : 'stop-tracks',
```

Comment:
@szuperaz, we should check if this makes sense once you are back. This defaulted to `disable-tracks` for the mic, and `stop-tracks` for the camera. I don't know the reasons behind this.

I tested with Web, and now the mic indicator goes away once both, mic and speaking-while-muted notifications are disabled. I left the previous default for React Native.